lib/addons/src/index.js

```diff
@@ -52,7 +53,16 @@ export class AddonStore {
   }
 
   addPanel(name, panel) {
-    this.panels[name] = panel;
+    // supporting legacy addons, which have not migrated to the active-prop
+    const original = panel.render;
```

Comment:
Does this technique work reliably? What if panel is a stateless functional component?

Perhaps we can check for `propTypes`?

Reply:
the `render` should be just a function as per the current docs. Sure in react a function === component, but neither we nor anyone in the ecosystem has put propTypes on that function.

Class-based components as render here, are not supported, not before of after this refactor.

Reply:
Can't we just check if `panel` is a function, and use it as `original` if so? Then we'd support SFCs too.

> Sure in react a function === component, but neither we nor anyone in the ecosystem has put propTypes on that function.

Not sure what you mean here.

If the user is using es6 classes (as our docs current use) or a SFC, they may have `panel.propTypes = { active: ... }`. Especially *if we tell them to in the new docs!*.

I would have thought a check of `panel.propTypes && !panel.propTypes.active` is a much better indicator of not using the new API than the fairly unreliable string check of the function body.